src/logic/graphFrames.ts

```diff
@@ -1,19 +1,95 @@
-import { Frame, DissasembledGraph, Algorithm } from './commonInterfaces';
+import {
+  Frame,
+  DisassembledGraph,
+  Algorithm,
+  Edge,
+  TraversalRecord,
+  Paths,
+  GraphIndex,
+} from './commonInterfaces';
+
+interface FrameWithEdges {
+  activeEdges: Array<Edge>,
+  inactiveEdges: Array<Edge>,
+};
 
 export default function createFrames(
-  dissasembledGraph: DissasembledGraph,
+  dissasembledGraph: DisassembledGraph,
   algorithm: Algorithm,
 ):
   Array<Frame>
 {
-  return [ // Temporary fixed result
-    { //frame 1
-      inactiveEdges: [],
-      activeEdges: [dissasembledGraph[1][0]],
-    },
-    { //frame 2
-      inactiveEdges: [dissasembledGraph[1][0]],
-      activeEdges: [],
-    },
-  ]
+  const edgeFrames = createFramesWithEdges(algorithm);
+  return convertToIndexes(dissasembledGraph, edgeFrames);
+}
+
+export function createFramesWithEdges(from: Algorithm) {
+  let frames: Array<FrameWithEdges> = [];
+  for (let tr of from.traversalHistory) {
+    frames.push(createFrameForNode(frames));
+    frames.push(...createFramesForLookups(tr, frames));
+  }
+  return frames;
+}
+
+function createFrameForNode(frames: Array<FrameWithEdges>): FrameWithEdges {
+  const lastFrame = frames[frames.length - 1] || { activeEdges: [], inactiveEdges: [] };
+  return {
+    activeEdges: [],
+    inactiveEdges: lastFrame.activeEdges.concat(lastFrame.inactiveEdges),
+  };
+}
+
+function createFramesForLookups(
+  tr: TraversalRecord,
+  frames: Array<FrameWithEdges>,
+):
+  Array<FrameWithEdges>
+{
+  const nodeFrames: Array<FrameWithEdges> = [
+    frames[frames.length - 1] || { activeEdges: [], inactiveEdges: [] },
+  ];
+  for (let toNode of tr.lookups) {
+    const lastFrame = nodeFrames[nodeFrames.length - 1];
+    nodeFrames.push({
+      activeEdges: [new Edge(tr.node, toNode)],
+      inactiveEdges: lastFrame.activeEdges.concat(lastFrame.inactiveEdges),
+    });
+  }
+  nodeFrames.shift() //First frame was repeated
+  return nodeFrames;
+}
+
+function convertToIndexes(
+  graph: DisassembledGraph,
+  edgeFrames: Array<FrameWithEdges>,
+):
+  Array<Frame>
+{
+  const frames: Array<Frame> = [];
+  const graphEdges = graph[GraphIndex.EDGES];
+  for (let eFrame of edgeFrames) {
+    frames.push({
+      activeEdges: convertEdgeArrayToIdexes(eFrame.activeEdges, graphEdges),
+      inactiveEdges: convertEdgeArrayToIdexes(eFrame.inactiveEdges, graphEdges),
+    })
+  }
+  return frames
+}
+
+function convertEdgeArrayToIdexes(
+  edgeArray: Array<Edge>,
+  graphEdges: Array<Edge>,
+):
+  Array<number>
+{
+  const indexes: Set<number> = new Set();
+  for (let edge of edgeArray) {
+    indexes.add(findEdgeIndex(edge, graphEdges));
+  }
+  return Array.from(indexes);
+}
+
+function findEdgeIndex(edge: Edge, edgeArray: Array<Edge>): number {
```

Comment:
Don't take it personal, code review is just  to idealize the quality,  but I think the chain of functions here is too "long". I would suggest using ES6 class next time or rethink organizing layers of abstractions somehow (I'm not saying the problem here is easy tho)

Wrapping up, for now let's leave it as it is.

Reply:
I don't really see problem with long function chains. It's just naming tasks. Layers of abstraction probably could be thought out better. Idk, it seems clear to me `operating on arrays -> operating on single array -> operating on single element`, but I'm the author.